workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/CommandMetaData.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    public struct CommandContext<T>
+    {
+        public Entity SendingEntity;
+        public T Request;
+        public object Context;
+
+        public CommandContext(Entity sendingEntity, T request, object context)
+        {
+            SendingEntity = sendingEntity;
+            Request = request;
+            Context = context;
+        }
+    }
+
+    public class CommandMetaData
+    {
+        private readonly Dictionary<uint, Dictionary<uint, ICommandMetaDataStorage>> componentIdToCommandIdToStorage =
+            new Dictionary<uint, Dictionary<uint, ICommandMetaDataStorage>>();
+
+        private List<(uint ComponentId, uint CommandId, uint InternalRequestId)> requestsToRemove =
```

Comment:
Can we make a lightweight struct for these rather than a 3 item tuple? Can be private to this class ofc